sdk/protocol-manager-poc/src/implementation/makerPlugin.ts

```diff
@@ -0,0 +1,221 @@
+                    maxSupply: TokenAmount.createFrom({token: collateralToken, amount: Number.MAX_SAFE_INTEGER.toString()}),
+                    liquidationPenalty: Percentage.createFrom({ percentage: dogRes.liquidationPenalty.toNumber() }),
+                    // apy: Percentage.createFrom({ percentage: 0 }),
+                }
+            }
+
+            const debts = {
+                [quoteToken.address.value]: {
+                    token: quoteToken,
+                    price: await ctx.priceService.getPrice({baseToken: quoteToken, quoteToken: collateralToken }),
+                    priceUSD: await ctx.priceService.getPriceUSD(quoteToken),
+                    rate: Percentage.createFrom({ percentage: stabilityFee.times(100).toNumber() }),
+                    totalBorrowed: TokenAmount.createFrom({token: quoteToken, amount: vatRes.normalizedIlkDebt.times(vatRes.debtScalingFactor).toString()}),
+                    debtCeiling: TokenAmount.createFrom({token: quoteToken, amount: vatRes.debtCeiling.toString()}),
+                    debtAvailable: TokenAmount.createFrom({token: quoteToken, amount:  vatRes.debtCeiling.minus(vatRes.normalizedIlkDebt.times(vatRes.debtScalingFactor)).toString()}),
+                    dustLimit: TokenAmount.createFrom({token: quoteToken, amount: vatRes.debtFloor.toString()}),
+                    originationFee: Percentage.createFrom({ percentage: 0 })
+                }
+            }
+
+            return {
+                type: PoolType.Lending,
+                poolId: makerPoolId,
+                protocol: makerPoolId.protocol,
+                baseCurrency: poolBaseCurrencyToken,
+                collaterals,
+                debts
+            }
+        },
+        getPositionId: (positionId: string): IPositionId => {
+            return positionId as IPositionId
+        },
+        getPosition: async (positionId: IPositionId): Promise<Position> => {
+            throw new Error(`Not implemented ${positionId}`)
+        },
+        _validate: function (candidate: unknown): asserts candidate is MakerPoolId {
+            const ProtocolNameEnum = z.nativeEnum(ProtocolName);
+            const IlkTypeEnum = z.nativeEnum(ILKType);
+            const ChainInfoType = z.object({
+                name: z.string(),
+                chainId: z.number()
+            })
+
+            const SparkPoolIdSchema = z.object({
+                protocol: z.object({
+                    name: ProtocolNameEnum,
+                    chainInfo: ChainInfoType
+                }),
+                ilkType: IlkTypeEnum
+            });
+
+            const parseResult = SparkPoolIdSchema.safeParse(candidate);
+            if (!parseResult.success) {
+                const errorDetails = parseResult.error.errors.map(error => `${error.path.join('.')} - ${error.message}`).join(', ');
+                throw new Error(`Candidate is not the correct shape: ${errorDetails}`);
```

Comment:
worth adding maker to the error messago for context: Candidate is not a correct MakerPoolId